### DISCUSSION TITLE
Keep proxycommand from sending the SSH user as a principal.

### DISCUSSION THREAD

**Discussion by flatheadmill** [smallstep/certificates#900] [General]
I've configured SSH to use `~/.ssh/authorized_principals` so that a user `fred.flinstone@slate.com` can log in using `fred@mysql.state.com`. When the user key expires, `step ssh proxycommand` will send a signing request with the principal `fred` which is rejected with ...

```
The request was forbidden by the certificate authority: ssh certificate principals does not match - got [fred], want [fredflinstone fredflinstone fred.flinstone@slate.com].
```

I tell Fred to just use `step ssh login` for now.

Is there a way to tell `step ssh proxycommand` to just use the email returned from the OIDC token?

I'm aware that I can solve this by writing an alternative to `step ssh proxycommand`. Just want to make sure I'm not missing anything.

**Comment by flatheadmill** [comment]
Realize this is more of a `step-cli` issue. Sorry. Should I move it?

**Reply by maraino** [reply]
Hi @flatheadmill, Are you using the open-source version of step-ca?

I believe you're, looking at the code I don't see a way right now to behave exactly as `step ssh login`. The change should be simple, although it might require a custom configuration of `~/.step/ssh/config`.

For the moment, I think the best solution is to use `step ssh login` once per day, it's possible to replace `step ssh proxycommand` with a bash script that works properly for that case, but it will be more costly than running `step ssh login` once per day.

If you want to adventure on the source code, perhaps replacing this like
https://github.com/smallstep/cli/blob/b6adb572306d49d207f6d76fb982374351505817/command/ssh/proxycommand.go#L136
```golang
principals := []string{subject}
```
with something like this 
```golang
if subject != "" {
	principals := []string{subject}
}
```

and passing `""` as the first argument to proxycommand instead of `%r` might do the trick.

**Reply by flatheadmill** [reply]
I did read through `proxycommand.go` before posting. I could make this change, but I could also work on it as a PR. Would this be something that would be accepted as a PR?

**Reply by maraino** [reply]
> I did read through proxycommand.go before posting. I could make this change, but I could also work on it as a PR. Would this be something that would be accepted as a PR?

@flatheadmill I think so, I think it will work like `step ssh login` with that change.

**Comment by Janhouse** [comment]
Related to https://github.com/smallstep/cli/pull/807

**Comment by maraino** [comment]
@flatheadmill, @Janhouse This PR might fix this issue https://github.com/smallstep/certificates/pull/1206
